shared/src/metabase/mbql/schema.cljc

```diff
@@ -638,8 +646,23 @@
 (defclause ^{:requires-features #{:date-extraction}} get-second
   datetime DateTimeExpressionArg)
 
+(def ^:private ArithmeticDateTimeUnit
+  (s/named
+   (apply s/enum #{:default :second :minute :hour :day :week :month :quarter :year})
```

Comment:
1. Should `:millisecond` be in here as well? Since some of the code above is written to support it.
2. I think we should disallow `:default` here. How does one add a `1 :default`? It doesn't make sense. We should require an explicit unit. When we use `:default` in other places it basically means "use this column at it's current resolution, don't apply truncation/bucketing"... doesn't really apply here.

```suggestion
   (apply s/enum #{:millisecond :second :minute :hour :day :week :month :quarter :year})
```